--- a/raid10.c
+++ b/raid10.c
@@ -4386,3 +4386,5 @@
 	lim.max_write_zeroes_sectors = 0;
+#ifdef HAVE_MAX_HW_WZEROES_UNMAP_SECTORS
 	lim.max_hw_wzeroes_unmap_sectors = 0;
+#endif
 	lim.logical_block_size = mddev->logical_block_size;
--- a/raid1.c
+++ b/raid1.c
@@ -3187,3 +3187,5 @@
 	lim.max_write_zeroes_sectors = 0;
+#ifdef HAVE_MAX_HW_WZEROES_UNMAP_SECTORS
 	lim.max_hw_wzeroes_unmap_sectors = 0;
+#endif
 	lim.logical_block_size = mddev->logical_block_size;
