--- a/raid10.c
+++ b/raid10.c
@@ -3935,7 +3935,9 @@
 
 	md_init_stacking_limits(&lim);
 	lim.max_write_zeroes_sectors = 0;
+#ifdef HAVE_MAX_HW_WZEROES_UNMAP_SECTORS
 	lim.max_hw_wzeroes_unmap_sectors = 0;
+#endif
 	lim.logical_block_size = mddev->logical_block_size;
 	lim.io_min = mddev->chunk_sectors << 9;
 	lim.chunk_sectors = mddev->chunk_sectors;
--- a/raid1.c
+++ b/raid1.c
@@ -3205,7 +3205,9 @@
 
 	md_init_stacking_limits(&lim);
 	lim.max_write_zeroes_sectors = 0;
+#ifdef HAVE_MAX_HW_WZEROES_UNMAP_SECTORS
 	lim.max_hw_wzeroes_unmap_sectors = 0;
+#endif
 	lim.logical_block_size = mddev->logical_block_size;
 	lim.features |= BLK_FEAT_ATOMIC_WRITES;
 	err = mddev_stack_rdev_limits(mddev, &lim, MDDEV_STACK_INTEGRITY);
