--- a/raid5.c
+++ b/raid5.c
@@ -7790,7 +7790,9 @@
 	lim.features |= BLK_FEAT_RAID_PARTIAL_STRIPES_EXPENSIVE;
 	lim.discard_granularity = stripe;
 	lim.max_write_zeroes_sectors = 0;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 18, 0)
 	lim.max_hw_wzeroes_unmap_sectors = 0;
+#endif
 	mddev_stack_rdev_limits(mddev, &lim, 0);
 	rdev_for_each(rdev, mddev)
 		queue_limits_stack_bdev(&lim, rdev->bdev, rdev->new_data_offset,
