--- a/md.c
+++ b/md.c
@@ -2020,8 +2020,10 @@
 
 		mddev->max_disks =  (4096-256)/2;
 
+#ifdef HAVE_MDP_SB1_LOGICAL_BLOCK_SIZE
 		if (!mddev->logical_block_size)
 			mddev->logical_block_size = le32_to_cpu(sb->logical_block_size);
+#endif
 
 		if ((le32_to_cpu(sb->feature_map) & MD_FEATURE_BITMAP_OFFSET) &&
 		    mddev->bitmap_info.file == NULL) {
@@ -2217,7 +2219,9 @@
 	sb->chunksize = cpu_to_le32(mddev->chunk_sectors);
 	sb->level = cpu_to_le32(mddev->level);
 	sb->layout = cpu_to_le32(mddev->layout);
+#ifdef HAVE_MDP_SB1_LOGICAL_BLOCK_SIZE
 	sb->logical_block_size = cpu_to_le32(mddev->logical_block_size);
+#endif
 	if (test_bit(FailFast, &rdev->flags))
 		sb->devflags |= FailFast1;
 	else
